py/sentry_streams/user_functions/function_template.py

```diff
@@ -0,0 +1,79 @@
+from abc import ABC, abstractmethod
+from typing import Any, Generic, Optional, TypeVar
+
+InputType = TypeVar("InputType")
+OutputType = TypeVar("OutputType")
+IntermediateType = TypeVar("IntermediateType")
+
+
+class AggregationBackend(ABC, Generic[OutputType]):
+    """
+    A storage backend that is meant to store windowed aggregates. Configurable
+    to the type of storage.
+    """
+
+    @abstractmethod
+    def flush_aggregate(self, aggregate: OutputType) -> None:
+        """
+        Flush a windowed aggregate to storage. Takes in the output from
+        the Accumulator.
+        """
+
+
+class Accumulator(ABC, Generic[InputType, IntermediateType, OutputType]):
+    """
+    The standard Accumulator template.
+    Define these functions to build a custom
+    Accumulator for aggregation.
```

Comment:
This will be the entry point for user to stateful streaming. 
Let's expand on this docstring explaining how it is supposed to be used and providing an example.
When you provide more examples for the API we will not need to have an example in the docstring.